text/0000-increase-build-plan-flexibility.md

```diff
@@ -0,0 +1,121 @@
+# Meta
+[meta]: #meta
+- Name: Increase Build Plan Flexibility
+- Start Date: 2020-05-28
+- Author(s): nebhale
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes removing the `version` key as a top-level key to properly reflect its importance in the broader buildpack ecosystem while continuing to allow buildpacks that do manage version metadata collaboration between buildpack to do so via the existing metadata mechanism.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the [build plan](https://github.com/buildpacks/spec/blob/master/buildpack.md#build-plan-toml) and the related [buildpack plan](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpack-plan-toml) are manifestations of the whiteboard pattern.  This pattern allows buildpacks to read and write arbitrary data both for themselves between detect and build times and between different buildpacks at build time.  This flexibility is _really_ important for innovation given that we cannot predict today, all of the novel ways buildpacks would want to communicate with themselves or other buildpacks.
+
+The one place where this flexibility is constrained is that `version` has been promoted to be a top-level key.  Given that so many buildpacks do not require `version` to be set in their interactions with other buildpacks, and the fact that in practice, there are only a very small number of actors (the original source code, the application developer explicitly) that can reasonably request a particular version, this elevation of the key reduces flexibility.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The `version` key should be removed as a top-level entry in both the build plan and buildpack plan.  Buildpacks that need to coordinate versions from other parties should use an agreed upon metadata key (the conventional, but not required, choice is `version`) to replicate their existing functionality.  The BOM schema will be updated to remove `version` as a top-level key as well, to match its relation to the buildpack plan.
+
+The `lifecycle` is required to implement temporary compatibility with the previous build plan and buildpack plan structure by converting the `version` key in `metadata` into and out of a top-level key at the appropriate locations.
```

Comment:
The only potential issue would be if the `version` key already exists in `metadata`, but that is an edge case. Would we want to error there?